packages/governance/src/committeeRegistrar.js

```diff
@@ -1,49 +1,71 @@
 // @ts-check
 
-import { Far } from '@agoric/marshal';
-import { makeNotifierKit } from '@agoric/notifier';
 import { E } from '@agoric/eventual-send';
-import { makeStore } from '@agoric/store';
+import { Far } from '@agoric/marshal';
+import { makeSubscriptionKit } from '@agoric/notifier';
 import { allComparable } from '@agoric/same-structure';
+import { makeStore } from '@agoric/store';
+import { natSafeMath } from '@agoric/zoe/src/contractSupport/index.js';
+
+import { makeHandle } from '@agoric/zoe/src/makeHandle';
+import { QuorumRule } from './question.js';
+
+const { ceilDivide } = natSafeMath;
 
 // Each CommitteeRegistrar represents a particular set of voters. The number of
 // voters is visible in the terms.
+//
+// This contract creates an electorate that is not visible to observers. There
+// may be uses for such a structure, but it is not appropriate for elections
+// where the set of voters needs to be known, unless the contract is used in a
+// way that makes the distribution of voter facets visible.
+
+/** @type {ContractStartFn} */
 const start = zcf => {
-  // Question => { voter, publicFacet }
+  /**
+   * @typedef {Object} QuestionRecord
+   * @property {ERef<VoteCounterCreatorFacet>} voteCap
+   * @property {VoteCounterPublicFacet} publicFacet
+   */
+
+  /** @type {Store<Handle<'Question'>, QuestionRecord>} */
   const allQuestions = makeStore('Question');
-  const { notifier, updater } = makeNotifierKit();
+  const { subscription, publication } = makeSubscriptionKit();
   const invitations = [];
```

Comment:
This is how I discovered the problem with the type of `getVoterInvitations`.

```suggestion
  /** @type { Promise<Invitation>[] } */
  const invitations = [];
```

In fact, initializing `invitations` to `[]` and later mutating it in a loop made me wonder if it's mutated in response to incoming messages. It's not. It might be too clever / obscure, but I prefer the following style in place of the for loop; `tsc` infers the type without an explicit annotation in this case:

```js
  const invitations = harden(
    [...Array(committeeSize).keys()].map(makeCommitteeVoterInvitation),
  );
```

Reply:
I'm not a fan of `[...Array(committeeSize).keys()].map(...)`, but `const invitations = harden(...)` is enough of an improvement that I've switched.